pkg/virtual/apiexport/controllers/apireconciler/apiexport_apireconciler_reconcile.go

```diff
@@ -57,29 +62,45 @@ func (c *APIReconciler) reconcile(ctx context.Context, apiExport *apisv1alpha1.A
 	oldSet := c.apiSets[apiDomainKey]
 	c.mutex.RUnlock()
 
-	// collect APIResourceSchemas
-	schemaIdentity := map[string]string{}
-
-	apiResourceSchemas := make([]*apisv1alpha1.APIResourceSchema, 0, len(apiExport.Spec.LatestResourceSchemas)+len(internalapis.Schemas))
-
-	// TODO(ncdc): switch to "ThingPermissionClaim" and remove this
-	for _, schema := range internalapis.Schemas {
-		shallow := *schema
-		shallow.ClusterName = logicalcluster.From(apiExport).String()
-		apiResourceSchemas = append(apiResourceSchemas, &shallow)
+	// Get schemas and identities for base api export
+	apiResourceSchemas, schemaIdentity, err := c.getSchemasFromAPIExport(apiExport)
+	if err != nil {
+		return err
 	}
 
-	for _, schemaName := range apiExport.Spec.LatestResourceSchemas {
-		apiResourceSchema, err := c.apiResourceSchemaLister.Get(clusters.ToClusterAwareKey(logicalcluster.From(apiExport), schemaName))
-		if err != nil && !apierrors.IsNotFound(err) {
-			return err
-		}
-		if apierrors.IsNotFound(err) {
-			klog.V(3).Infof("APIResourceSchema %s in APIExport %s|% not found", schemaName, apiExport.Namespace, apiExport.Name)
-			continue
+	for _, pc := range apiExport.Spec.PermissionClaims {
+		if pc.IdentityHash != "" {
+			exports, err := c.apiExportIndexer.ByIndex(indexAPIExportByIdentity, pc.IdentityHash)
+			if err != nil {
+				return err
+			}
+			for _, export := range exports {
+				e, ok := export.(*apisv1alpha1.APIExport)
+				e = e.DeepCopy()
+				if !ok {
+					return fmt.Errorf("invalid type from apiExportIndexer got: %T, expected %T", export, apiExport)
+				}
+				schemas, schemaIDs, err := c.getSchemasFromAPIExport(e)
+				if err != nil {
+					return err
+				}
+				apiResourceSchemas = append(apiResourceSchemas, schemas...)
+				for k, v := range schemaIDs {
+					// Override the permission claims that should be used. for this schema identity
+					v.Spec.PermissionClaims = apiExport.Spec.PermissionClaims
```

Comment:
Instead of doing this, we could pass in the permissions claims. I wonder what is better

Reply:
Am lost here. Why change the owning APIExport?